### DISCUSSION TITLE
Rebus with HealthCheck - AzureServiceBus

### DISCUSSION THREAD

**Discussion by rafilsk0** [rebus-org/Rebus#1133] [General]
Guys, Have anyway to configure a healthCheck using Rebus configurations?

**Comment by mookid8000** [comment]
I am not sure I understand your question... is there some part of the configuration you are in doubt about?

**Comment by vilinski** [comment]
I would be also interested in how to, or whether it were possible to configure the health check in a web app. 
It should indicate whether the message bus is operational, can receive messages, transport is healthy, queue sizes are under limit.
Usualy it is done using `IHealthChecker`

```csharp
serviceCollection
    .AddHealthChecks()
    .AddRebus()

static IHealthCheckBuilder AddRebus(this IHealthCheckBuilder builder, string connStringOrAndOtherArgs) 
{
    ....
    // TODO what can be done?
    return builder;
}
```
In my case the transport is over RabbitMQ, but should be probably similar in any case

**Comment by mookid8000** [comment]
Doesn't there exist health check for Azure Service Bus / RabbitMQ / (...) that does what you need?

**Reply by vilinski** [reply]
sure there is one at least for rabbitmq. But rebus abstracts the underlying platform for me, and I thought adding hc should work through one of rebus nuget too, for example Rebus.ServiceProvider
But if youre asking and I didn't found anything documented, I think we are on our own here.

**Reply by mookid8000** [reply]
Sorry, but yeah you're on your own here 🙂 

And yes, Rebus abstracts away the underlying queueing systems. But each queueing system has additional features and their own extensibility packages (e.g. for health checking a particular queueing system in ASP.NET Core), and supporting them via Rebus would be a LOT of work! 

Consider RabbitMQ health checks – if I understand things correctly, that comes in a NuGet package that depends on the RabbitMQ client and ASP.NET Core? – if Rebus had to integrate with that, I would either have to bring in a dependency on that NuGet package, thus making Rebus.RabbitMq depend on ASP.NET Core, or I would have to create a new, separate project for Rebus.RabbitMQ.AspNetCoreHealthChecks, or something like that.

But why stop there? Why not also Rebus.RabbitMq.Prometheus, Rebus.RabbitMq.OpenTelemetry, Rebus.RabbitMq.ElmahIo, etc.? You can probably imagine the explosion of NuGet packages 😅 

It would be a lot of work, and it would probably only be marginally easier to use compared to bringing in the RabbitMQ+ASP.NET Core health check NuGet package and configuring it, so that's why it is like it is.
